src/ApiResponseHelpers.php

```diff
@@ -33,17 +36,23 @@ public function respondNotFound(
     /**
      * @param array|Arrayable|JsonSerializable|null $contents
      */
-    public function respondWithSuccess($contents = []): JsonResponse
+    public function respondWithSuccess($contents = null): JsonResponse
     {
-        $contents = $this->morphToArray($contents);
+        $contents = $this->morphToArray($contents) ?? [];
 
         $data = [] === $contents
-            ? ['success' => true]
+            ? $this->_api_helpers_defaultSuccessData
             : $contents;
 
         return $this->apiResponse($data);
     }
 
+    public function setDefaultSuccessResponse(?array $content = null)
```

Comment:
We could add a `self` or `static` phpdoc, however from my tests this doesn't help PHPStorm realize that this is fluent method returning class using this trait. I were unable to make it understand that.
But it realizes it returns object with trait methods so it's good enough. If anyone has idea how to make this better then give this a shot.

Reply:
I'll take a look at that, usually `self` has always been enough. However, as this is via a trait, as you've discovered, that may change.

Reply:
PHPStorm was nagging about no return type and suggested `self`. I've added that for good measure.